Apps/Src/SendCarCAN.c

```diff
@@ -1,20 +1,57 @@
-#include "os.h"
 #include "Tasks.h"
-#include "bsp.h"
+#include "common.h"
 #include "CANbus.h"
 #include "CAN_Queue.h"
+#include "Pedals.h"
+#include "Minions.h"
+#include "Contactors.h"
 
+#define carMSGID 0x581
 
-/**
- * @brief This Task is a basic queue consumer that takes CAN Messages out of the OS Queue and sends them
-*/
+typedef union DATAMSG{
+    struct{
+        uint64_t accelPedal : 8;
+        uint64_t brakePedal : 8;
+        uint64_t switchBitmap : 16;
+        uint64_t lightBitmap : 8;
+        uint64_t contactorBitmap : 8;
+        uint64_t reserved : 16;
+    };
+    uint64_t val;
+}DATAMSG_t;
 
 void Task_SendCarCAN(void *p_arg){
-    CANMSG_t msg;
+    CANMSG_t motorMSG;
+    CANMSG_t carMSG;
+    DATAMSG_t data;
 
     while (1) {
-        CAN_Queue_Pend(&msg);
-        CANbus_Send(msg.id,msg.payload,CAN_BLOCKING); //send message
+        // Send motor msg
+        CAN_Queue_Pend(&motorMSG);
+        CANbus_Send(motorMSG.id, motorMSG.payload, CAN_BLOCKING);
+
+        // Get pedal information
+        data.accelPedal = Pedals_Read(ACCELERATOR);
+        data.brakePedal = Pedals_Read(BRAKE);
+
+        // Get switch information
+        for(int i = 0; i < NUM_SWITCHES; i++){
+            bool switchState = (bool)Switches_Read(i);
```

Comment:
Casting States to bools is technically unsafe. Use a ternary:
`Switches_Read(i)==ON?true:false`